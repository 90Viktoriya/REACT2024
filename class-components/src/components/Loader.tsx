import { Component } from 'react';
import styles from './components.module.css';

export class Loader extends Component {
  render() {
    return <div className={styles.loader}></div>;
  }
}
