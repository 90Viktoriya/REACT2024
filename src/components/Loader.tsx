import styles from './components.module.css';

export function Loader() {
  return <div className={styles.loader}></div>;
}
