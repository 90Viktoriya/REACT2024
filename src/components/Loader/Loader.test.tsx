import { describe, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Loader } from './Loader';

describe('Given Loader component', () => {
  it('when rendered, should match snapshot', () => {
    const { asFragment } = render(<Loader />);

    expect(asFragment()).toMatchSnapshot();
  });
});
