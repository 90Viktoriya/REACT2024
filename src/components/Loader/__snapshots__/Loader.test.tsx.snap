// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given Loader component > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_loader_bf668d"
  />
</DocumentFragment>
`;
